aiida_common_workflows/generators/generator.py

```diff
@@ -3,13 +3,31 @@
 import abc
 import copy
 
-from aiida import engine
+from aiida import engine, orm
 from ..protocol import ProtocolRegistry
 from .spec import InputGeneratorSpec
 
 __all__ = ('InputGenerator',)
 
 
+def recursively_check_stored_nodes(obj):
+    """
+    This class is used to copy the `dict` containig the arguments passed to `get_builder`
+    following the rules:
+    1) if the value is a node and it is stored, the same value is returned
+    2) if the value is something else, it is deepcopied
+    This function takes care of applying the copy rules also to nested namespaces.
+    :param obj: whatever object, in the `InputGenerator` it receives the `dict` containig
+                `get_builder` arguments.
+    :return: a copy of obj following the rules above
+    """
+    if isinstance(obj, dict):
+        return {k: recursively_check_stored_nodes(v) for k, v in obj.items()}
+    if isinstance(obj, orm.Data) and obj.is_stored:
```

Comment:
Actually, I think we should not limit this to just `Data` nodes. It is possible that `ProcessNodes` are passed as well, for example the `reference_workchain`, and there `deepcopy` will even raise. So we should also just return it.